Automattic/wp-calypso
# Theme showcase: Refactor categories

Fixes https://github.com/Automattic/dotcom-forge/issues/3059

## Proposed Changes

<img width="749" alt="Screenshot 2023-07-27 at 16 35 47" src="https://github.com/Automattic/wp-calypso/assets/1233880/338f874d-0cac-46d8-9a1d-3ea6f0aa1d09">

- Rename the current "Recommended" tab to "All" and move it to the end of the list.
- Add a new "Recommended" tab that is just curated themes (`menu_order` > 0).
- The "All" tab is selected when searching without a specific category.

## Testing Instructions

- Use the Calypso live link below
- Go to Appearance > Themes
- Make sure the "Recommend" tab only includes curated themes (e.g. it shouldn't include "Twenty Sixteen").
- Make sure the "All" tab includes all themes (e.g. it should include "Twenty Sixteen").
- Make sure the "All" tab is selected when searching without a specific category.
- Also check the theme showcase for a Jetpack and an Atomic site and make sure there are no regressions, particularly in the "My themes" tab.
- Also check the logged out theme showcase and make sure there are no regressions.
